lib/commands/actions.js

```diff
@@ -169,33 +172,62 @@ commands.fingerprint = async function (fingerprintId) {
   await this.adb.fingerprint(fingerprintId);
 };
 
+async function storeScreenshotDataWithAdbShell (adb, opts, localFile) {
+  let pngDir = opts.androidScreenshotPath || '/data/local/tmp/';
+  const png = path.posix.resolve(pngDir, 'screenshot.png');
+  let cmd = ['/system/bin/rm', `${png};`, '/system/bin/screencap', '-p', png];
+  await adb.shell(cmd);
+  await adb.pull(png, localFile);
+}
+
+async function storeScreenshotDataWithAdbExecOut (adb, localFile) {
+  let {stdout} = await exec(adb.executable.path, ['exec-out', '/system/bin/screencap -p']);
+  if (!_.isString(stdout) || new Buffer(stdout).toString('hex', 0, 4) !== PNG_MAGIC) {
+    throw new Error(`Command output '${stdout}' is not a valid PNG file`);
```

Comment:
Printing the `stdout` here might lead to some weird logs:
```
info AndroidDriver Cannot get screenshot data with 'adb exec-out' because of 'Command output '�PNG
info AndroidDriver
IHDR���d� IDATx���w|\՝���uYݶ,�I�\po����6Y`Y�nB�$)d��
                                                   1I(aC��c�e\���{�%۲��{���!��A�-Y#�~=?�;s��g.�y�9��sm���n�.G0�@`
 �N�:��#G*((�㾲�2?~\���4�`HHH����>�9sF|��$�O�>�7o�BBB���*55���=b<x��5)$$��>��&��Զ������G}T?��O�e��z뭚2e��|�M��[�x�6m�Ԫ��1B111JHHhuM�n�7
info AndroidDriver [�˥��r���J�k�߿�կu��
                                      Է~�����d��4l�0IҶm�t�}��*����b5���6����6���_�rF����:�I�&Y����GTQQ�$��'�l�6��]E``����U[[��Ǐ����tI��V����O~��Çז-[$IS�Li����
info AndroidDriver ���\���Z������Խ{w9�މ��o�Y
                                            .Ԁ4cƌ&��޽��~�i]{��
info AndroidDriver
```